project-base/storefront/hooks/comparison/useComparison.tsx

```diff
@@ -9,58 +8,43 @@ import {
 import { getUserFriendlyErrors } from 'helpers/errors/friendlyErrorMessageParser';
 import { useTypedTranslationFunction } from 'hooks/typescript/useTypedTranslationFunction';
 import { useCurrentUserData } from 'hooks/user/useCurrentUserData';
-import { Dispatch, SetStateAction, useEffect, useMemo, useState } from 'react';
+import { useEffect, useState } from 'react';
 import { usePersistStore } from 'store/zustand/usePersistStore';
 
-export const useHandleCompare = (
-    productUuid: string,
-): {
-    isProductInComparison: boolean;
-    handleProductInComparison: () => void;
-    handleRemoveAllFromComparison: () => void;
-    isPopupCompareOpen: boolean;
-    setIsPopupCompareOpen: Dispatch<SetStateAction<boolean>>;
-    getComparisonProducts: () => ComparedProductFragmentApi[];
-} => {
+export const useComparison = () => {
     const t = useTypedTranslationFunction();
-    const [isPopupCompareOpen, setIsPopupCompareOpen] = useState(false);
     const { isUserLoggedIn } = useCurrentUserData();
-    const [isProductInComparison, setIsProductInComparison] = useState(false);
     const [, addProductToComparison] = useAddProductToComparisonMutationApi();
     const [, removeProductFromComparison] = useRemoveProductFromComparisonMutationApi();
     const [, cleanComparison] = useCleanComparisonMutationApi();
-    const productsComparisonUuid = usePersistStore((store) => store.productsComparisonUuid);
+    const comparisonUuid = usePersistStore((store) => store.productsComparisonUuid);
     const updateUserState = usePersistStore((store) => store.updateUserState);
-    const [result] = useComparisonQueryApi({
-        variables: { comparisonUuid: productsComparisonUuid },
+    const [isPopupCompareOpen, setIsPopupCompareOpen] = useState(false);
+    const [pause, setPause] = useState(true);
+
+    const [{ data, fetching }] = useComparisonQueryApi({
+        variables: { comparisonUuid },
         requestPolicy: 'network-only',
+        pause,
```

Comment:
WIth these "custom-working queries" I kind of started leaning towards not using the auto-generated hooks and rather writing the logic ourselves. Do you think it could simplify this as well?

Reply:
I don't know what exactly it should simplify. Let's discuss this on call.

Reply:
If you had the query logic in a useEffect itself, you would not need this pause hack. We can discuss it, yeah